### ISSUE TITLE
GTX 1080 Ti Cuda Launch Failed

### ISSUE THREAD

**Issue by adanaila** [tensorflow/tensorflow#10852]
Please go to Stack Overflow for help and support:

http://stackoverflow.com/questions/tagged/tensorflow

If you open a GitHub issue, here is our policy:

1. It must be a bug or a feature request.
2. The form below must be filled out.
3. It shouldn't be a TensorBoard issue. Those go [here](https://github.com/tensorflow/tensorflow/issues).

**Here's why we have that policy**: TensorFlow developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

------------------------

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
Linux Ubuntu 16.04
- **TensorFlow installed from (source or binary)**:
Source
- **TensorFlow version (use command below)**:
('v1.1.0-rc2-9-gb7f85a7', '1.1.0-rc1')
- **Bazel version (if compiling from source)**:
Bazel Release 0.5.0

- **CUDA/cuDNN version**:
Cuda 8.0 cuDNN 8.0
- **GPU model and memory**:
GTX 1080Ti - 11GB memory
- **Exact command to reproduce**:

bazel-bin/inception/flowers_train \
  --train_dir="${TRAIN_DIR}" \
  --data_dir="${DATA_DIR}" \
  --pretrained_model_checkpoint_path="${MODEL_PATH}" \
  --fine_tune=True \
  --initial_learning_rate=0.001 \
  --input_queue_memory_factor=2

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

[tf_env.txt](https://github.com/tensorflow/tensorflow/files/1088718/tf_env.txt)

You can obtain the TensorFlow version with

python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

### Describe the problem
Describe the problem clearly here. Be sure to convey here why it's a bug in TensorFlow or a feature request.

When fine tuning inception v3 model using the flowers data script, on custom data, during retraining I see the following error:

2017-06-19 23:46:32.549857: step 5300, loss = 1.40 (61.9 examples/sec; 0.517 sec/batch)
2017-06-19 23:46:39.510009: step 5310, loss = 1.43 (64.2 examples/sec; 0.499 sec/batch)
2017-06-19 23:46:44.667435: step 5320, loss = 1.55 (64.3 examples/sec; 0.498 sec/batch)
2017-06-19 23:46:46.691129: E tensorflow/stream_executor/cuda/cuda_driver.cc:1067] could not synchronize on CUDA context: CUDA_ERROR_LAUNCH_FAILED :: No stack trace available
2017-06-19 23:46:46.691129: E tensorflow/stream_executor/cuda/cuda_event.cc:49] Error polling for event status: failed to query event: CUDA_ERROR_LAUNCH_FAILED
2017-06-19 23:46:46.691162: F tensorflow/core/common_runtime/gpu/gpu_event_mgr.cc:203] Unexpected Event status: 1
2017-06-19 23:46:46.691165: F tensorflow/core/common_runtime/gpu/gpu_util.cc:370] GPU sync failed
./start_image.sh: line 60: 27406 Aborted                 (core dumped) bazel-bin/inception/flowers_train --train_dir="${TRAIN_DIR}" --data_dir="${DATA_DIR}" --pretrained_model_checkpoint_path="${MODEL_PATH}" --fine_tune=True --initial_learning_rate=0.001 --input_queue_memory_factor=2

I changed the num classes and num examples per epoch as follows:
 7    def num_classes(self):
      8      """Returns the number of classes in the data set."""
      9 -    return 5
     10 +    return 21
     11  
     12    def num_examples_per_epoch(self):
     13      """Returns the number of examples in the data subset."""
     14      if self.subset == 'train':
     15 -      return 3170
     16 +      return 6342
     17      if self.subset == 'validation':
     18 -      return 500
     19 +      return 1576

### Source code / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem.

**Comment by gunan** [comment]
tf_env_collect script says you do not have cuDNN installed.
Note that CUDA and cuDNN need to be installed separately.

Please install cuDNN following all instructions from NVIDIA and retry.

**Comment by adanaila** [comment]
I have made sure that the cudnn libraries are located in the /local/usr/cuda/lib64 and /local/usr/cuda/include:

If they were not, I could not build tensorflow with cuda support

cookie@ceres:~/work/machine_learning/tensorflow/tools$ ls -l /usr/local/cuda/lib64/libcudnn*
-rwxr-xr-x 1 root root 148M Jun 21 07:31 /usr/local/cuda/lib64/libcudnn.so
-rwxr-xr-x 1 root root 148M Jun 21 07:31 /usr/local/cuda/lib64/libcudnn.so.6
-rwxr-xr-x 1 root root 148M Jun 21 07:31 /usr/local/cuda/lib64/libcudnn.so.6.0.21
-rw-r--r-- 1 root root 138M Jun 21 07:31 /usr/local/cuda/lib64/libcudnn_static.a

Is there anything else I can check or do? The flowers_train script runs for multiple interations, before it crashes each time with different errors, as below:

2017-06-21 10:32:20.376515: step 17080, loss = 0.96 (62.9 examples/sec; 0.509 sec/batch)
2017-06-21 10:32:25.431246: step 17090, loss = 0.96 (62.7 examples/sec; 0.511 sec/batch)
2017-06-21 10:32:30.495980: step 17100, loss = 0.89 (62.9 examples/sec; 0.509 sec/batch)
2017-06-21 10:32:33.929246: E tensorflow/stream_executor/cuda/cuda_event.cc:49] Error polling for event status: failed to query event: CUDA_ERROR_ILLEGAL_ADDRESS
2017-06-21 10:32:33.929280: F tensorflow/core/common_runtime/gpu/gpu_event_mgr.cc:203] Unexpected Event status: 1

**Comment by gunan** [comment]
How about your `LD_LIBRARY_PATH`?
For running tensorflow, your `LD_LIBRARY_PATH` needs to include directories to your cuda, cudnn libcudart and libcupti so file directories.

**Comment by adanaila** [comment]
I checked all the libraries and they are located in the LD_LIBRARY_PATH:

cookie@ceres:~/work/machine_learning/flowers_training$ echo $LD_LIBRARY_PATH
/usr/local/cuda/extras/CUPTI/lib64:/usr/lib/x86_64-linux-gnu/:/usr/local/cuda-8.0/lib64

I just reran the model:
2017-06-21 21:58:50.719064: step 550, loss = 1.14 (62.5 examples/sec; 0.512 sec/batch)
2017-06-21 21:58:55.847020: step 560, loss = 1.22 (62.6 examples/sec; 0.511 sec/batch)
2017-06-21 21:59:00.008349: E tensorflow/stream_executor/cuda/cuda_event.cc:49] Error polling for event status: failed to query event: CUDA_ERROR_ILLEGAL_ADDRESS
2017-06-21 21:59:00.008379: F tensorflow/core/common_runtime/gpu/gpu_event_mgr.cc:203] Unexpected Event status: 1
2017-06-21 21:59:00.008384: F tensorflow/stream_executor/cuda/cuda_dnn.cc:1961] failed to enqueue convolution on stream: CUDNN_STATUS_BAD_PARAM
./train_flowers.sh: line 27: 13497 Aborted                 (core dumped) bazel-bin/inception/flowers_

Anything else I can check?

**Comment by adanaila** [comment]
@gunan Anything else I can do to attempt to root cause this? I've tried different drivers, different cuda versions/ cudnn versions all with the same results.

This is even happening when running flowers_train on the flowers data with no changes to the model.

**Comment by gunan** [comment]
Looks like with the update to `LD_LIBRARY_PATH` error mode has changed to `CUDNN_STATUS_BAD_PARAM`. When searching through old issues, it looks like the cause of this error is using TF built with another cudnn version. based on your comments, if you built on the same machine as you are runningg this should not be the case.
By any chance, do you have multiple versions of cudnn installed?

**Comment by adanaila** [comment]
Thank you for your time in helping me with this.

I only have 1 version of cuda installed, cuda-8.0

My /usr/local looks as follows:

> lrwxrwxrwx  1 root root    8 Jun 21 07:27 cuda -> cuda-8.0
> drwxr-xr-x 14 root root 4.0K Jun 21 07:27 cuda-8.0


My /usr/local/cuda-8.0 looks as follows:

> drwxr-xr-x  3 root root 4.0K Jun 21 07:27 bin
> drwxr-xr-x  5 root root 4.0K Jun 21 07:27 doc
> drwxr-xr-x  5 root root 4.0K Jun 21 07:27 extras
> drwxr-xr-x  8 root root 4.0K Jun 21 07:27 libnsight
> drwxr-xr-x  7 root root 4.0K Jun 21 07:27 libnvvp
> drwxr-xr-x  3 root root 4.0K Jun 21 07:27 nvml
> drwxr-xr-x  7 root root 4.0K Jun 21 07:27 nvvm
> drwxr-xr-x 11 root root 4.0K Jun 21 07:27 samples
> drwxr-xr-x  3 root root 4.0K Jun 21 07:27 share
> drwxr-xr-x  2 root root 4.0K Jun 21 07:27 src
> drwxr-xr-x  3 root root 4.0K Jun 21 07:27 targets
> drwxr-xr-x  2 root root 4.0K Jun 21 07:27 tools
> -rw-r--r--  1 root root  365 Jan 26 15:48 LICENSE
> -rw-r--r--  1 root root  365 Jan 26 15:48 README
> lrwxrwxrwx  1 root root   28 Jan 26 15:51 include -> targets/x86_64-linux/include
> lrwxrwxrwx  1 root root   24 Jan 26 15:51 lib64 -> targets/x86_64-linux/lib
> -rw-r--r--  1 root root   20 Jan 26 15:48 version.txt

Inside the lib64 symlink, I see the following:

> <snip>....
> -rwxr-xr-x 1 root root  148M Jun 21 07:31 libcudnn.so
> -rwxr-xr-x 1 root root  148M Jun 21 07:31 libcudnn.so.6
> -rwxr-xr-x 1 root root  148M Jun 21 07:31 libcudnn.so.6.0.21
> <snip>....

LD_LIBRARY_PATH is: 

> /usr/local/cuda/extras/CUPTI/lib64:/usr/local/cuda-8.0/lib64

Just to make sure, I rebuilt tensorflow and re-ran the model a few more times with the same result

> 2017-06-23 22:13:27.662981: step 16670, loss = 1.04 (64.7 examples/sec; 0.494 sec/batch)
> 2017-06-23 22:13:28.589187: E tensorflow/stream_executor/cuda/cuda_driver.cc:1067] could not synchronize on CUDA context: CUDA_ERROR_ILLEGAL_ADDRESS :: No stack trace available
> 2017-06-23 22:13:28.589246: F tensorflow/core/common_runtime/gpu/gpu_util.cc:370] GPU sync failed
> 2017-06-23 22:13:28.589187: E tensorflow/stream_executor/cuda/cuda_event.cc:49] Error polling for event status: failed to query event: CUDA_ERROR_ILLEGAL_ADDRESS
> 2017-06-23 22:13:28.589273: F tensorflow/core/common_runtime/gpu/gpu_event_mgr.cc:203] Unexpected Event status: 1

**Comment by gunan** [comment]
Just to make sure, could you locate `libcublas.so`, `libcuda.so`, `libcufft.so` and `libcurand.so` on your system?

**Comment by adanaila** [comment]
Everything _but_ libcuda.so is in /usr/local/cuda/lib64

libcuda.so is I can find a few:

> /usr/local/cuda-8.0/targets/x86_64-linux/lib/stubs/libcuda.so
> /usr/lib/i386-linux-gnu/libcuda.so
> /usr/lib/x86_64-linux-gnu/libcuda.so


Looking for installed cuda packages I see:

> cookie@ceres:~$ apt list --installed | grep cuda
> 
> WARNING: apt does not have a stable CLI interface. Use with caution in scripts.
> 
> cuda/unknown,now 8.0.61-1 amd64 [installed]
> cuda-8-0/unknown,now 8.0.61-1 amd64 [installed,automatic]
> cuda-command-line-tools-8-0/unknown,now 8.0.61-1 amd64 [installed,automatic]
> cuda-core-8-0/unknown,now 8.0.61-1 amd64 [installed,automatic]
> cuda-cublas-8-0/unknown,now 8.0.61-1 amd64 [installed,automatic]
> cuda-cublas-dev-8-0/unknown,now 8.0.61-1 amd64 [installed,automatic]
> cuda-cudart-8-0/unknown,now 8.0.61-1 amd64 [installed,automatic]
> cuda-cudart-dev-8-0/unknown,now 8.0.61-1 amd64 [installed,automatic]
> cuda-cufft-8-0/unknown,now 8.0.61-1 amd64 [installed,automatic]
> cuda-cufft-dev-8-0/unknown,now 8.0.61-1 amd64 [installed,automatic]
> cuda-curand-8-0/unknown,now 8.0.61-1 amd64 [installed,automatic]
> cuda-curand-dev-8-0/unknown,now 8.0.61-1 amd64 [installed,automatic]
> cuda-cusolver-8-0/unknown,now 8.0.61-1 amd64 [installed,automatic]
> cuda-cusolver-dev-8-0/unknown,now 8.0.61-1 amd64 [installed,automatic]
> cuda-cusparse-8-0/unknown,now 8.0.61-1 amd64 [installed,automatic]
> cuda-cusparse-dev-8-0/unknown,now 8.0.61-1 amd64 [installed,automatic]
> cuda-demo-suite-8-0/unknown,now 8.0.61-1 amd64 [installed,automatic]
> cuda-documentation-8-0/unknown,now 8.0.61-1 amd64 [installed,automatic]
> cuda-driver-dev-8-0/unknown,now 8.0.61-1 amd64 [installed,automatic]
> cuda-drivers/unknown,now 375.26-1 amd64 [installed,automatic]
> cuda-license-8-0/unknown,now 8.0.61-1 amd64 [installed,automatic]
> cuda-misc-headers-8-0/unknown,now 8.0.61-1 amd64 [installed,automatic]
> cuda-npp-8-0/unknown,now 8.0.61-1 amd64 [installed,automatic]
> cuda-npp-dev-8-0/unknown,now 8.0.61-1 amd64 [installed,automatic]
> cuda-nvgraph-8-0/unknown,now 8.0.61-1 amd64 [installed,automatic]
> cuda-nvgraph-dev-8-0/unknown,now 8.0.61-1 amd64 [installed,automatic]
> cuda-nvml-dev-8-0/unknown,now 8.0.61-1 amd64 [installed,automatic]
> cuda-nvrtc-8-0/unknown,now 8.0.61-1 amd64 [installed,automatic]
> cuda-nvrtc-dev-8-0/unknown,now 8.0.61-1 amd64 [installed,automatic]
> cuda-repo-ubuntu1604-8-0-local-ga2/now 8.0.61-1 amd64 [installed,local]
> cuda-runtime-8-0/unknown,now 8.0.61-1 amd64 [installed,automatic]
> cuda-samples-8-0/unknown,now 8.0.61-1 amd64 [installed,automatic]
> cuda-toolkit-8-0/unknown,now 8.0.61-1 amd64 [installed,automatic]
> cuda-visual-tools-8-0/unknown,now 8.0.61-1 amd64 [installed,automatic]
> libcuda1-375/xenial-updates,xenial-security,now 375.66-0ubuntu0.16.04.1 amd64 [installed,automatic]

**Comment by adanaila** [comment]
This was likely a hardware issue. Replacing the 1080Ti with an identical card, resolves the issue all software staying the same.

The issue can be closed. Many thanks for your time!

**Comment by t0mst0ne** [comment]
@adanaila  may I ask which brand 1080Ti you use when error occurred ? 
I met the same situation on EVGA 1080ti ftw3 , not sure if the hardware problem , thanks for your help.

**Comment by adanaila** [comment]
@t0mst0ne I had a founder's card, it either crashed the system or resulted in the above errors.

**Comment by alexantoinefortin** [comment]
@adanaila , I ran into this issue when overclocking my card.
If you still have your old card, it might be worth it to give it another try. Perhaps, you could underclock it, to go easy on it.
